src/main/java/ac/grim/grimac/player/GrimPlayer.java

```diff
@@ -554,14 +574,25 @@ public CompensatedInventory getInventory() {
         return checkManager.getInventory();
     }
 
-    public List<Double> getPossibleEyeHeights() { // We don't return sleeping eye height
-        if (getClientVersion().isNewerThanOrEquals(ClientVersion.V_1_14)) { // Elytra, sneaking (1.14), standing
+    public double[] getPossibleEyeHeights() { // We don't return sleeping eye height
+        if (getClientVersion().isNewerThanOrEquals(ClientVersion.V_1_14)) {
             final float scale = (float) compensatedEntities.getSelf().getAttributeValue(Attributes.GENERIC_SCALE);
-            return Arrays.asList(0.4 * scale, 1.27 * scale, 1.62 * scale);
-        } else if (getClientVersion().isNewerThanOrEquals(ClientVersion.V_1_9)) { // Elytra, sneaking, standing
-            return Arrays.asList(0.4, 1.54, 1.62);
-        } else { // Only sneaking or standing
-            return Arrays.asList((double) (1.62f - 0.08f), (double) (1.62f));
+            if (this.isGliding || this.isSwimming) {
+                return new double[]{0.4 * scale, 1.62 * scale, 1.27 * scale}; // Elytra, standing, sneaking (1.14)
+            } else if (this.isSneaking) {
+                return new double[]{1.27 * scale, 1.62 * scale, 0.4 * scale}; // sneaking (1.14), standing, Elytra
+            } else {
+                return new double[]{1.62 * scale, 1.27 * scale, 0.4 * scale}; // standing, sneaking (1.14), Elytra
+            }
+        } else if (getClientVersion().isNewerThanOrEquals(ClientVersion.V_1_9)) {
+            if (this.isGliding || this.isSwimming) {
+                return possibleEyeHeights[2]; // Elytra
+            }
+        }
+        if (this.isSneaking) {
+            return possibleEyeHeights[1];
+        } else {
+            return possibleEyeHeights[0];
         }
```

Comment:
This can be one line
```suggestion
        return this.isSneaking ? possibleEyeHeights[1] : possibleEyeHeights[0];
```

Reply:
Appreciate it. Do you have any ideas on how I can go about detecting changes to scale and caching it efficiently?